sa/storage-authority.go

```diff
@@ -74,22 +67,26 @@ func statusIsPending(status core.AcmeStatus) bool {
 	return status == core.StatusPending || status == core.StatusProcessing || status == core.StatusUnknown
 }
 
-func existingPending(tx *gorp.Transaction, id string) bool {
+func existing(tx *gorp.Transaction, query string, id interface{}) bool {
 	var count int64
-	_ = tx.SelectOne(&count, "SELECT count(*) FROM pendingAuthorizations WHERE id = :id", map[string]interface{}{"id": id})
-	return count > 0
+	err := tx.SelectOne(&count, query, map[string]interface{}{"id": id})
+	return err != nil || count > 0
```

Comment:
Good point.  The way it is now has the unpleasant consequence that the SA will keep generating unique IDs forever if there's no database there.  I've modified this to log the error, and also added explicit limits on the "guess and check" ID generation loops.